### PR TITLE
Fix #12791

### LINKED ISSUE #12791: AttributeTypecastBehavior uses its owner before attachment
Inside a behavior `$this->owner` is `NULL` at the moment of the initialization.

[Here](https://github.com/yiisoft/yii2/blob/master/framework/behaviors/AttributeTypecastBehavior.php#L197) `get_class($this->owner)` always evaluates to `yii\behaviors\AttributeTypecastBehavior` since `$this->owner` is `NULL`.

[Here](https://github.com/yiisoft/yii2/blob/master/framework/behaviors/AttributeTypecastBehavior.php#L272) a `Call to a member function getValidators() on null` error occurs for the same reason.

### PR DESCRIPTION
**Pull request by rmrevin**
| Q | A |
| --- | --- |
| Is bugfix? | yes |
| New feature? | no |
| Breaks BC? | no |
| Tests pass? | yes |
| Fixed issues | #12791 |

### CODE REVIEW

File: framework/behaviors/AttributeTypecastBehavior.php [hunk 1]
```diff
@@ -189,17 +189,25 @@ public static function clearAutoDetectedAttributeTypes()
...
-            }
-            $this->attributeTypes = self::$autoDetectedAttributeTypes[$ownerClass];
+            $this->reinitAutoDetectedAttributeTypes();
+        }
+    }
+
+    /**
+     * Reinitialization typecast attributes for model
+     */
+    public function reinitAutoDetectedAttributeTypes()
```
Comment (cebe) [inline framework/behaviors/AttributeTypecastBehavior.php]: I don't think there is the need to create a public method for this.

**Review by cebe — CHANGES_REQUESTED** [review]

File: framework/behaviors/AttributeTypecastBehavior.php (hunk 1 above)
Comment (cebe) [inline framework/behaviors/AttributeTypecastBehavior.php]: as far as I see, you can call attach() in the test directly.

Comment (rmrevin) [inline framework/behaviors/AttributeTypecastBehavior.php]: @cebe fixed

**Comment by klimov-paul** [comment]
Resolved by commit d4df72a2a4d83ce2f4ae4fef4ca252834d6a319b

**Pull request closed without merging**